benchmark/intl_stream_benchmark.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:math';
+
+// Import BenchmarkBase class.
+import 'package:benchmark_harness/benchmark_harness.dart';
+
+const int asciiZeroCodeUnit = 2;
+const int listlength = 10000;
+
+// Create a new benchmark by extending BenchmarkBase
+class NewMethod extends BenchmarkBase {
+  late String string;
+  late String string2;
+  int zeroDigit = 15;
+  NewMethod() : super('New');
+
+  static void main() {
+    NewMethod().report();
+  }
+
+  @override
+  void setup() => string = String.fromCharCodes(Iterable.generate(
+        listlength,
+        (index) => Random().nextInt(80) + zeroDigit,
+      ));
+  // The benchmark code.
+  @override
+  void run() {
+    var codeUnits = string.codeUnits;
+    string2 = String.fromCharCodes(List.generate(
+      codeUnits.length,
+      (index) => codeUnits[index] - zeroDigit + asciiZeroCodeUnit,
+      growable: false,
+    ));
+
+    // var codeUnits = string.codeUnits;
+    // string2 = String.fromCharCodes(codeUnits
+    //     .map((c) => c - zeroDigit + asciiZeroCodeUnit)
+    //     .toList(growable: false));
+  }
+}
+
+// Create a new benchmark by extending BenchmarkBase
+class OldMethod extends BenchmarkBase {
+  late String string;
+  late String string2;
+  int zeroDigit = 15;
+  OldMethod() : super('Old');
+
+  static void main() {
+    OldMethod().report();
+  }
+
+  @override
+  void setup() => string = String.fromCharCodes(Iterable.generate(
+        listlength,
+        (index) => Random().nextInt(80) + zeroDigit,
+      ));
+  // The benchmark code.
+  @override
+  void run() {
+    var oldDigits = string.codeUnits;
+    var newDigits = List<int>.filled(string.length, 0);
+    for (var i = 0; i < string.length; i++) {
+      newDigits[i] = oldDigits[i] - zeroDigit + asciiZeroCodeUnit;
+    }
+    string2 = String.fromCharCodes(newDigits);
+  }
+}
+
+void main() {
```

Comment:
At least on my machine, the new method runs quite a bit faster. I would remove the old method references from this file when merging, this is just for you to compare.

Reply:
Please compare native (AOT) and web (dart2js) to ensure the changes improve both platforms.

Reply:
`dart compile exe <script>` should give you an AOT build.

`dart compile js <script>` will give you the dart2js (our release web compiler) output. In order to run that, you'd need to add a minimal html file in this directory referencing that js file. `dart create --template=web foo` will create a simple dart web hello world app; I imagine you could copy and update its html file here.

We do have a full benchmarking CI for Dart. It has some setup involved; I myself wouldn't worry about putting this change under continuous benchmarking (but it would be good to become familiar w/ the system, for when we want to make sure that some performance aspects don't regress).

Reply:
Thanks, that worked!

AOT:  ~7% faster
dart2js: ~29% faster
JIT: ~39% faster